src/Elastic.Apm.StartupHook.Loader/Loader.cs

```diff
@@ -40,40 +44,72 @@ private static string AssemblyDirectory
 		/// </summary>
 		public static void Initialize()
 		{
-			var agentLibsToLoad = new[] { "Elastic.Apm", "Elastic.Apm.Extensions.Hosting", "Elastic.Apm.AspNetCore", "Elastic.Apm.EntityFrameworkCore", "Elastic.Apm.SqlClient", "Elastic.Apm.GrpcClient", "Elastic.Apm.Elasticsearch" };
-			var agentDependencyLibsToLoad = new[] { "System.Diagnostics.PerformanceCounter", "Microsoft.Diagnostics.Tracing.TraceEvent", "Newtonsoft.Json", "Elasticsearch.Net" };
+			_agentLoadContext = new AgentLoadContext(AssemblyDirectory, Logger);
+
+			foreach (var libToLoad in AgentLoadContext.AgentDependencyLibsToLoad)
+				LoadAssembly(libToLoad);
+			foreach (var libToLoad in AgentLoadContext.AgentLibsToLoad)
+				LoadAssembly(libToLoad);
+
+			AssemblyLoadContext.Default.Resolving += (context, assemblyName) =>
+			{
+				if (AgentLoadContext.AgentDependencyLibsToLoad.Contains(assemblyName.Name)
+					|| AgentLoadContext.AgentLibsToLoad.Contains(assemblyName.Name))

```

Comment:
Should we make both `AgentDependencyLibsToLoad` and `AgentLibsToLoad` to be `HashSet<string>` for this op?